sessions/forecast-ensembles.qmd

```diff
@@ -48,127 +52,230 @@ The source file of this session is located at `sessions/forecast-ensembles.qmd`.
+  mutate(h = row_number()) |> 
+  ungroup() |> 
+  ## this ensures that the output is a fable object
+  as_fable(response = "wili", distribution = wili)
+```
+
+We plot all the forecasts, with one plot for each cross-validation split of the data.
 
 ```{r}
-gen_time_pmf <- make_gen_time_pmf()
-ip_pmf <- make_ip_pmf()
-onset_df <- simulate_onsets(
-  make_daily_infections(infection_times), gen_time_pmf, ip_pmf
-)
-head(onset_df)
+autoplot(
+  cv_forecasts,
+  flu_data_tscv |> filter(epiweek >= as.Date("2017-07-01")),
+  alpha = 0.5
+) +
+  facet_wrap(~.split) +
+  theme(legend.position = "bottom")
 ```
 
+# Forecast representation and format
+
+One important thing about working with forecast data in practice is that following some standard ways of representing individual forecasts can make it easier to do a wider range of analyses with a broad array of models.
+While `fable` is good for some fairly simple modeling, we will want and need more complexity and a more flexible representation of forecast data.
+In this section, we will start using hubverse-style data for forecast output formats.
+
+::: callout-tip
+## What is the hubverse?
+
+The [hubverse](https://hubverse.io/) is a project designed to improve public health outcomes through collaborative modeling.
+Hubverse data standards and software support collaborative modeling "hubs" around the world, including in Europe, Paraguay, Australia, the US and Canada.
+The standards were developed for and by folks doing epidemiological forecasting and projections, but they are general and could be used by practitioners in other fields as well.
+
+Two key hubverse terms to keep in mind are:
+
+-   A **modeling task** is a specific thing that is being predicted. It is represented by a specific set of variables that the hubverse calls "task-id variables". In the example that we have been working with, the modeling task is defined by the date the forecast is made (`"forecast_date"`) and the horizon being predicted (`"h"`). Note that we have also been predicting for a given location (`"region"`), although we've only been predicting for one location so far. Also, each prediction refers to a specific epiweek (`"target_epiweek"`) although this is entirely defined by `"forecast_date"` and `"h"`, so in this way we can think of it as a "derived" variable.
+-   A **output type** refers to the way that a specific prediction is represented in data. Common output types are mean, median, quantile and sample. The hubverse documentation site has [more detail on these and other output types](https://docs.hubverse.io/en/latest/user-guide/model-output.html#details-about-model-output-column-specifications).
 :::
 
+A strength of the hubverse data standards is that it facilitates comparison and synthesis of models that are made with a wide variety of different modeling tools.
+If you can format your output data according to the simple standards, then your model could be part of a modeling hub.
+
+## Reformatting model output
```

Comment:
this is a lot of information about data formatting perhaps it can go in a collapsed call out?